From 7d83ef0736f1c8569f8df29eaeb9040bfc327f84 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 6 May 2024 14:09:14 +0200
Subject: GCC: returning an std::optional of a value requires move operator

In rekor.cc, returning an std::optional of the locally created
LogEntry requires the move operator.

Bug: 40565911
Change-Id: Ib40ff1ce60ccc8401e98ff233df1702c6cab2524
---
 device/fido/enclave/verify/rekor.cc | 3 ++-
 device/fido/enclave/verify/rekor.h  | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/device/fido/enclave/verify/rekor.cc b/device/fido/enclave/verify/rekor.cc
index a2085e7b9eec6..5288d330117e8 100644
--- a/device/fido/enclave/verify/rekor.cc
+++ b/device/fido/enclave/verify/rekor.cc
@@ -67,7 +67,8 @@ LogEntry::LogEntry(std::string body,
       verification(std::move(verification)) {}
 LogEntry::LogEntry() = default;
 LogEntry::~LogEntry() = default;
-LogEntry::LogEntry(LogEntry& log_entry) = default;
+LogEntry::LogEntry(const LogEntry& log_entry) = default;
+LogEntry::LogEntry(LogEntry&& log_entry) = default;
 
 void LogEntry::RegisterJSONConverter(
     base::JSONValueConverter<LogEntry>* converter) {
diff --git a/device/fido/enclave/verify/rekor.h b/device/fido/enclave/verify/rekor.h
index 4abd8db1d30b7..4dde2d2fe1da1 100644
--- a/device/fido/enclave/verify/rekor.h
+++ b/device/fido/enclave/verify/rekor.h
@@ -40,7 +40,8 @@ struct COMPONENT_EXPORT(DEVICE_FIDO) LogEntry {
            std::string log_id,
            uint64_t log_index,
            std::optional<LogEntryVerification> verification);
-  LogEntry(LogEntry& log_entry);
+  LogEntry(const LogEntry& log_entry);
+  LogEntry(LogEntry&& log_entry);
   LogEntry();
   ~LogEntry();
 
-- 
2.40.1

